### PR TITLE
Migrate to TypeScript

### PR DESCRIPTION
**Pull request by g-plane**
I just open this PR for better reviewing. If there are any problems, everyone can push new commits directly, or merge this PR if all are well.

### CODE REVIEW

**Comment by yushijinhun** [comment]
I suggest to remove the `options` parameter in the constructor of `SkinViewer`. What do you think of this?

File: src/model.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import * as THREE from "three";
 
-function toFaceVertices(x1, y1, x2, y2, w, h) {
+// TODO move to a util class
```
[the remark this answers is not in the document]
Comment (Hacksore) [reply src/model.ts]: Fair point, remove the comments.

File: src/model.ts [hunk 2]
```diff
@@ -33,18 +38,31 @@ function setVertices(box, top, bottom, left, front, right, back) {
 	box.faceVertexUvs[0][11] = [back[0], back[1], back[2]];
 }
 
+// why is this a global constant?
```
[the remark this answers is not in the document]
Comment (Hacksore) [reply src/model.ts]: I've seen this in other three.js projects so not really sure what we would do other than a global.

**Comment by Hacksore** [comment]
@yushijinhun I find having the `options' param is a good thing though? Makes it easier to define almost everything you need in one instantiation of the `SkinViewer` object.

**Comment by Hacksore** [comment]
Also @g-plane I had added .browser to the rollup bundle as there were compiled issues when building. 

Did you try the building with your change first?

**Comment by g-plane** [comment]
What compilation issues?

**Comment by Hacksore** [comment]
Seems like you mitigated the previous compile issue I was having so you can disregard.

**Comment by yushijinhun** [comment]
@Hacksore

You are right. I just found it a bit difficult to add a type definition for `options` (maybe because I'm a ts newcomer). Could you do this?

File: src/model.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yushijinhun) [reply src/model.ts]: It is just used in this file and is not exported. No need to worry so much.

**Comment by Hacksore** [comment]
@yushijinhun I think typically it's best to make an interface for the options so that IntelliSense will autocomplete the options. 

However, since this is targeted for the browser I'm not sure this will translate very well 😃 

So something like this.
```typescript
interface ObjectTest {
	slim: boolean,
	optional?: string
}

function configTest(object: ObjectTest){
	console.log(object)
}

configTest({
	slim: true,
	optional: "This is optional"
})
```

We might just have to make it a gerenric type `any`.

**Comment by Hacksore** [comment]
Is the plan to merge this?

**Comment by g-plane** [comment]
That's true.

**Pull request merged**